### PR TITLE
Makefile updates (EXTRA_LDFLAGS, dos2unix)

### PR DESCRIPTION
**Pull request by mikee47**
Move linker flags into their appropriate Components.

### CODE REVIEW

File: Sming/Makefile [hunk 1]
```diff
@@ -140,6 +140,14 @@ cs: ##Apply coding style to all core files
 		clang-format -i -style=file $$FILE; \
 	done
 
+ALL_FILES = $(filter %.cpp %.h %.c %.mk %.sh %.cmd,$(shell $(GIT) ls-files))
+
+.PHONY: dos2unix
+dos2unix: ##Make sure all our files use Unix line endings
+	@dos2unix $(ALL_FILES)
```
Comment (slaff) [inline Sming/Makefile]: Please, use `$(Q) dos2unix ...` instead of `@dos2unix`. Same for the line below.

Comment (mikee47) [reply Sming/Makefile]: OK, have done that.

**Comment by slaff** [comment]
@mikee47 Last words before merging this PR?

**Comment by mikee47** [comment]
@slaff All done, thanks

**Pull request merged**